components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImpl.java

```diff
@@ -50,15 +54,46 @@ public APIMgtAdminServiceImpl(APISubscriptionDAO apiSubscriptionDAO, PolicyDAO p
     }
 
     /**
-     * @see org.wso2.carbon.apimgt.core.api.APIMgtAdminService#getAPISubscriptions(int)
+     * @see APIMgtAdminService#getPoliciesByLevel(PolicyLevel)
+     */
+    @Override
+    public List<Policy> getPoliciesByLevel(PolicyLevel policyLevel) throws APIManagementException {
+        try {
+            return policyDAO.getPoliciesByLevel(policyLevel);
+
+        } catch (APIMgtDAOException e) {
+            String errorMessage = "Couldn't retrieve Throttle Policies with level: " + policyLevel.name();
+            log.error(errorMessage, e);
+            throw new APIManagementException(errorMessage, e, e.getErrorHandler());
```

Comment:
IMHO we should not; if we do so, we will lose the information set by the DAO (or the next lower) layer.